app/config.go

```diff
@@ -411,12 +411,13 @@ func (s *Server) SaveConfig(newCfg *model.Config, sendConfigChangeClusterMessage
 		return model.NewAppError("saveConfig", "app.save_config.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
 
-	if s.Metrics != nil {
-		if *s.Config().MetricsSettings.Enable {
-			s.Metrics.StartServer()
-		} else {
-			s.Metrics.StopServer()
+	if s.startMetrics && *s.Config().MetricsSettings.Enable {
+		if s.Metrics != nil {
+			s.Metrics.Register()
```

Comment:
Is register idempotent? or what is the same, is the router `Handle` method idempotent?

Reply:
Yes, it is. Register will use same `prometheus.Registry`.  As far as I see the `Handle` method will be idempotent in this case as we don't modify the route thereafter.